fix(*): general fixes

kaiarrowood opened the pull request:
### Summary
General fixes from the redesign

#### Changes made:
* Update `KTable` docs to remove sortable when not sorting
* `KTable` - fix client sort and add experimental `enableClientSort` prop
* `KtextArea`
  * Add prop `disableCharLimit`
  * Fix alignment of char limit text
  * Emit an event if the character limit is exceeded
  * Update docs / tests

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

adamdehaven commented on docs/components/textarea.md:
````diff
@@ -31,22 +31,38 @@ If the label is omitted it can be handled with another component, like **KLabel*
+<KTextArea :disableCharacterLimit="true" />
+
+```vue
+<KTextArea :disableCharacterLimit="true" />
````
Unless the standard is to explicitly show props with a `true` value, this could just be written as

``` html
<KTextArea disableCharacterLimit />

<!-- or -->

<KTextArea disable-character-limit />
```

kaiarrowood replied:
We don't really have a standard, and do a mix throughout the docs, but I went ahead and changed this instance.

adamdehaven reviewed (commented):
If we're providing a `characterLimit` prop, should we not be enforcing that character limit by limiting the text the user can enter into the field, or at least providing a way to hook into whether it's valid? 

It looks like this is more just for user feedback, but if I was using this component I would possibly assume that the character limit functionality would also trim the value of the field (e.g. anything over the `characterLimit` is trimmed from the end of the value).

At the least, I would think the component would provide a way to hook into whether or not the value is valid (i.e. if the value is within the `characterLimit` parameters), at least with a computed property that indicates whether the character limit has been violated, or some sort of `$emit`.

aanchalm01 commented on packages/KTextArea/KTextArea.vue:
```diff
@@ -118,28 +118,37 @@ export default {
-    color: var(--grey-500);
-  }
+  textarea.form-control {
+    font-family: 'Maison Neue';
```
```suggestion
    font-family: var(--font-family-sans);
```

kaiarrowood replied:
Got it

kaiarrowood commented:
> If we're providing a `characterLimit` prop, should we not be enforcing that character limit by limiting the text the user can enter into the field, or at least providing a way to hook into whether it's valid?
> 
> It looks like this is more just for user feedback, but if I was using this component I would possibly assume that the character limit functionality would also trim the value of the field (e.g. anything over the `characterLimit` is trimmed from the end of the value).
> 
> At the least, I would think the component would provide a way to hook into whether or not the value is valid (i.e. if the value is within the `characterLimit` parameters), at least with a computed property that indicates whether the character limit has been violated, or some sort of `$emit`.

Good idea @adamdehaven I've added a `char-limit-exceeded` event that is emitted when the character limit is exceeded and updated the docs.
